server/src/main/java/org/elasticsearch/index/mapper/MappedFieldType.java

```diff
@@ -148,7 +148,7 @@ public Object valueForDisplay(Object value) {
      * Returns true if the field is searchable.
      */
     public boolean isSearchable() {
-        return isIndexed;
+        return isIndexed || hasDocValues();
```

Comment:
I think this needs to be made in `ScaledFloatFieldType` by overwriting `isSearchable()` method.
Or maybe let `ScaledFloatFieldType` extend from `NumberFieldType`?